discovery-provider/src/tasks/index.py

```diff
@@ -351,355 +376,284 @@ def update_ursm_address(self):
-                    if track_ids:
-                        remove_cached_track_ids(redis, track_ids)
-                if playlist_state_changed:
-                    if playlist_ids:
-                        remove_cached_playlist_ids(redis, playlist_ids)
+            try:
+                session.commit()
                 logger.info(
-                    f"index.py | redis cache clean operations complete for block=${block_number}"
+                    f"index.py | session committed to db for block=${block_number}"
                 )
-            except IndexingError as err:
-                logger.info(
-                    f"index.py | Error in the indexing task at"
-                    f" block={err.blocknumber} and hash={err.txhash}"
+            except Exception as e:
+                # Use 'commit' as the tx hash here.
+                # We're at a point where the whole block can't be added to the database, so
+                # we should skip it in favor of making progress
+                blockhash = update_task.web3.toHex(block_hash)
+                indexing_error = IndexingError(
+                    "session.commit", block_number, blockhash, "commit", str(e)
                 )
-                set_indexing_error(
-                    redis, err.blocknumber, err.blockhash, err.txhash, err.message
+                create_and_raise_indexing_error(indexing_error, redis)
+            try:
+                # Check the last block's timestamp for updating the trending challenge
+                [should_update, date] = should_trending_challenge_update(
+                    session, latest_block_timestamp
                 )
-                confirm_indexing_transaction_error(
-                    redis, err.blocknumber, err.blockhash, err.txhash, err.message
+                if should_update:
+                    celery.send_task(
+                        "calculate_trending_challenges", kwargs={"date": date}
+                    )
+            except Exception as e:
+                # Do not throw error, as this should not stop indexing
+                logger.error(
+                    f"index.py | Error in calling update trending challenge {e}",
+                    exc_info=True,
                 )
-                raise err
-        try:
-            # Check the last block's timestamp for updating the trending challenge
-            [should_update, date] = should_trending_challenge_update(
-                session, latest_block_timestamp
-            )
-            if should_update:
-                celery.send_task("calculate_trending_challenges", kwargs={"date": date})
-        except Exception as e:
-            # Do not throw error, as this should not stop indexing
-            logger.error(
-                f"index.py | Error in calling update trending challenge {e}",
-                exc_info=True,
-            )
+            if skip_tx_hash:
+                clear_indexing_error(redis)
+
+        if changed_entity_ids_map:
```

Comment:
currently this is right after process_state_changes, should this be indented more?

Reply:
i checked and i believe this is at the correct indentation level. it runs once at the end of each block getting processed. i did realize i should reset changed_entity_ids_map to an empty dict at the end of this though so that's good! thanks for pointing this out.